### PR TITLE
fix(service/header): lazily load Store head

### LINKED ISSUE #457: [BUG]: store is always initialized
After restart, the header Store of a Node is always reinitialized even if the reference to the new latest head exists.

### PR DESCRIPTION
**Pull request by Wondertan**
Fixes #457 by making store head being lazy loaded and not on Start.

### CODE REVIEW

File: service/header/store.go [hunk 1]
```diff
@@ -140,10 +130,21 @@ func (s *store) Stop(ctx context.Context) error {
 
 func (s *store) Head(ctx context.Context) (*ExtendedHeader, error) {
 	head, err := s.GetByHeight(ctx, s.heightSub.Height())
-	if err == ErrNotFound {
+	if err == nil {
```
Comment (liamsi) [inline service/header/store.go:133]: ~~In the case `err != nil` the error might get overwritten by readHead and not handled a few lines below?~~
Ah, it's handled by the switch case.

File: service/header/store.go [hunk 2]
```diff
@@ -140,10 +130,21 @@ func (s *store) Stop(ctx context.Context) error {
...
+		return head, err
```
Comment (liamsi) [inline service/header/store.go:134]:
```suggestion
		return head, nil
```

**Review by liamsi — DISMISSED** [review]

**Comment by Wondertan** [comment]
The initialization flow:
1 Node creates Store
2 Node initializes Store
	2a Checks if there is any head
	2b If not - requests and writes requested head to Store
3 Node starts Store

The bug was with the fact that step 2a required step 3, as Head was only loaded on start. Now that head is lazy-loaded, 2a can understand if there is head without 3

File: service/header/store.go (hunk 2 above)
Comment (Wondertan) [reply service/header/store.go:134]: We that it is nil by the if case, but it doesn't hurt to say it is nil explicitly.

**Review by renaynay — APPROVED** [review]
MGMT

**Comment by Wondertan** [comment]
@renaynay, yeah, nice band 😄 
> MGMT

**Pull request merged**